lib/scraper_test.rb

```diff
@@ -30,17 +31,47 @@ def install_tasks
           it 'should contain the expected data' do
             # TODO: Make this configurable
             Dir['test/data/*.yml'].each do |file|
-              yaml_data = YAML.load_file(file).to_h
-              url = yaml_data[:url]
-              class_to_test = Object.const_get(yaml_data[:class])
-              VCR.use_cassette(File.basename(url)) do
-                response = class_to_test.new(response: Scraped::Request.new(url: url).response)
-                response.to_h.must_equal yaml_data[:to_h]
+              VCR.use_cassette(File.basename(file)) do
+                instructions = Instructions.new(file)
+                response = instructions.class_to_test
+                                       .new(response: Scraped::Request.new(url: instructions.url).response)
+                response.to_h.must_equal instructions.expected
               end
             end
           end
         end
       end
     end
   end
+
+  class Instructions
+    def initialize(filename)
+      @filename = filename
+    end
+
+    def url
+      test_data[:url] or raise "No url provided by #{filepath.basename}."
```

Comment:
Here, and throughout, the message might be better a "supplied in", rather than "provided by", to make it clearer that this is simply a field that we expect to see directly in the file, rather than something that could be deduced from it.